addons/default/visiosoft/cats-module/src/Category/Listener/CalculatedTotalForEditedAdCategory.php

```diff
@@ -0,0 +1,57 @@
+<?php namespace Visiosoft\CatsModule\Category\Listener;
+
+use Visiosoft\AdvsModule\Adv\Event\CreatedAd;
+use Visiosoft\AdvsModule\Adv\Event\EditedAdCategory;
+use Visiosoft\CatsModule\Category\Contract\CategoryRepositoryInterface;
+
+class CalculatedTotalForEditedAdCategory
+{
+    protected $categoryRepository;
+
+    public function __construct(CategoryRepositoryInterface $categoryRepository)
+    {
+        return $this->categoryRepository = $categoryRepository;
+    }
+
+    public function handle(EditedAdCategory $event)
+    {
+        $ad_detail = $event->getAdDetail()->toArray();
+        $before_editing_ad = $event->getBeforeEditingParams();
+
+        //Categories New Ad
+        $category_fields_new_ad = preg_grep('/^cat/i', array_keys($ad_detail));
```

Comment:
is there any better solutions?

Reply:
waiting for the solutions you know. @fatihalp